docs/trace/advanced/stratified-sampling-example/README.md

```diff
@@ -0,0 +1,103 @@
+# Stratified Sampling: An Example
```

Comment:
Are there plans to support Jaeger's remote sampling? It's also a form of stratified sampling, by the endpoint (span name).

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet/issues/4253 Didn't realize its something part of spec and we are missing! Opened an issue to track it.

Reply:
@yurishkuro I see this question as tangential, maybe. Jaeger Remote is absolutely desirable, but isn't there still a question about this **Exported, Unsampled** state of span that requires coordination between Sampler and Processor? Seems we've run into an OTel spec-level problem, you can't "just" be a Sampler to do the kinds of sampling we want to do, you have to be both a Sampler and a Processor. See also my remark here: https://github.com/open-telemetry/opentelemetry-dotnet/pull/4206#discussion_r1122662595

Reply:
@jmacd my point was that rather than providing a hypothetical example of how stratified sampling could be done (a perhaps valuable, but also very contrived example), supporting the Jaeger's remote sampling protocol achieves similar goal with a much higher, production-proven usefulness.

Jaeger's implementation of per-endpoint sampling mostly expects span name to be available on creation, although two SDKs support sampling based on late binding of span name. Those later two indeed have bigger implications on what "sampled" status means, by introducing a deferred sampling state. In case of Jaeger, that was managed as an interaction between Tracer and Sampler.

Reply:
Thanks @yurishkuro and @jmacd. I am reading more about Jaeger's Remote sampling now and I see how it is a form of stratified sampling by span name. While I agree the example in this PR is contrived, my current feeling is that it is a smaller and more self-contained example for a reader to get started on understanding the stratified sampling concept. So, personally I see a benefit of having this example in addition to .NET SDK's future implementation of Jaeger's remote sampling (as part of the issue #4253).  Please let me know if you disagree, and I am willing to be convinced otherwise.

Reply:
Thanks @yurishkuro. I think @kalyanaj's PR is a useful demonstration of how Samplers and Processors are bound together. I think we have convincing evidence that:
- OTel has no standard way to export export spans from unsampled contexts (except using `tracestate: ot=p:63` to signify zero adjusted count)
- users want to change span-export decisions based on latency and error status, which are late-arriving information
- the OTel Sampler API is limiting in the sense that you can't mix-and-match Processors and Samplers arbitrarily and expect to achieve tail sampling

While I admire the Jaeger remote sampling configuration and want to see SDK-configurable sampling, the evidence above shows that when we implement such a Sampler/Processor it ends up as a monolithic Sampler/Processor pair. Not only is it difficult to compose Samplers because they're tied up with Processors for tail sampling, [we've shown that it is difficult to compose samplers for head sampling that respects Span-to-Metrics accounting](https://github.com/open-telemetry/opentelemetry-specification/issues/2179). In other words, to implement the Jaeger remote sampler and support Span-to-Metrics, where every exported span has an accurate/unbiased adjusted count for itself, you will have a hard time re-using existing components.